src/rgw/rgw_rest_s3.h

```diff
@@ -873,6 +873,83 @@ static inline int valid_s3_bucket_name(const string& name, bool relaxed=false)
   return 0;
 }
 
+class s3select;
+class RGWS3Select : public RGWGetObj_ObjStore_S3
+{
+
+private:
+  s3select *s3select_syntax;
+  std::string m_last_line;
+  bool m_previous_line;
+
+public:
+  unsigned int chunk_number;
+  u_int64_t m_processed_bytes;
+
+  typedef enum
+  {
+    EVENT_TYPE,
+    CONTENT_TYPE,
+    MESSAGE_TYPE
+  } header_name_en_t;
+  static const char *header_name_str[3];
+
+  typedef enum
+  {
+    RECORDS,
+    OCTET_STREAM,
+    EVENT
+  } header_value_en_t;
+  static const char *header_value_str[3];
+
+  RGWS3Select();
+  virtual ~RGWS3Select();
+
+//encoding into big-endian
```

Comment:
why not use a library function here (htonl etc.) ? you do an extra copy here

Reply:
not sure about that , lets discuss it offline